### ISSUE TITLE
No package for Debian Bullseye

### ISSUE THREAD

**Issue by crozone** [dogecoin/dogecoin#1692]
### Describe the issue

The package freeze for Debian Bullseye is rapidly approaching and no `dogecoin` package exists in the Bullseye repository.

Attempt to install the dogecoin package on Debian Bullseye:

`sudo apt install dogecoin`

### Expected behaviour

The package is installed successfully and dogecoind and dogecoin-cli are installed.

### Actual behaviour

```
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package dogecoin
```

From what I can tell, these are the blockers that prevent master from building on Bullseye:

* Boost 1.74 requires `#include <boost/bind.hpp>` to be added to some source files:
  * https://github.com/dogecoin/dogecoin/issues/1654
  * PR https://github.com/dogecoin/dogecoin/pull/1655 fixes this for 1.14 (NOT yet merged due to failing build)

* httpserver.cpp requires `#include <deque>`
  * https://github.com/dogecoin/dogecoin/pull/1626 (Merged for 1.14, not for master)

* Support Berkeley Database 5.3 / libdb-dev 5.3.1

It would be great if a prebuilt deb package was available for Bullseye, to ease dogecoin daemon adoption.

**Comment by ReverseControl** [comment]
@crozone We do not target Debian's Bullseye specifically. You can build from source, or get the binaries here: https://github.com/dogecoin/dogecoin/releases. 

@patricklodder @rnicoll @michilumin I think we can close this one.

**Comment by ReverseControl** [comment]
@crozone Could you test with 1.14.4?

**Comment by crozone** [comment]
`./configure` gave me the standard "configure: error: Found Berkeley DB other than 5.1, required for portable wallets (--with-incompatible-bdb to ignore or --disable-wallet to disable wallet functionality)".

Debian is currently on `libdb-dev (5.3.1+nmu1)`. Since 5.3 is the same major version as 5.1 and the on-disk format appears to be the same, I opted for `./configure --with-incompatible-bdb`.

With that, 1.14.4 successfully builds on Debian 11.0, `Linux 5.10.0-8-amd64 #1 SMP Debian 5.10.46-4 (2021-08-03) x86_64 GNU/Linux`, GCC `gcc (Debian 10.2.1-6) 10.2.1 20210110`

Build steps:

```
./autogen.sh
./configure --with-incompatible-bdb
make
```

```
Options used to compile and link:
  with wallet   = yes
  with gui / qt = no
  with zmq      = yes
  with test     = yes
  with bench    = yes
  with upnp     = yes
  debug enabled = no
  werror        = no

  target os     = linux
  build os      =

  CC            = gcc
  CFLAGS        = -g -O2
  CPPFLAGS      =  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS
  CXX           = g++ -std=c++11
  CXXFLAGS      = -g -O2 -Wall -Wextra -Wformat -Wvla -Wformat-security -Wno-unused-parameter
  LDFLAGS       =
  ```

The build still had lots of implicit fallthrough warnings like:

```
tinyformat.h:707:21: warning: this statement may fall through [-Wimplicit-fallthrough=]
  707 |             out.setf(std::ios::uppercase);
      |             ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
```

Might want to consider fixing the fallthroughs or adding `-Wimplicit-fallthrough=0` to CXXFLAGS.

Besides that everything built well, I'm now moving on to testing the binary.

**Comment by crozone** [comment]
I've started testing on my node proper with `dogecoind` built with `./configure --disable-wallet --without-miniupnpc`

```
Options used to compile and link:
  with wallet   = no
  with gui / qt = no
  with zmq      = yes
  with test     = yes
  with bench    = yes
  with upnp     = no
  debug enabled = no
  werror        = no

  target os     = linux
  build os      =

  CC            = gcc
  CFLAGS        = -g -O2
  CPPFLAGS      =  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS
  CXX           = g++ -std=c++11
  CXXFLAGS      = -g -O2 -Wall -Wextra -Wformat -Wvla -Wformat-security -Wno-unused-parameter
  LDFLAGS       =
  ```
  
  So far everything looks good.

**Comment by ReverseControl** [comment]
@crozone Great! Note that #1655 and #1626 were both merged, and #1655 fixed #1654. When you are done testing let me know and we can close this one. Much thank you.

**Comment by crozone** [comment]
I can report that the node is working well. Looks like we're good to close!
